gordo_client/io.py

```diff
@@ -76,8 +78,11 @@ def _handle_response(resp: requests.Response, resource_name: Optional[str] = Non
         In case of network or IO errors
     """
     if 200 <= resp.status_code <= 299:
-        is_json = resp.headers["content-type"] == "application/json"
-        return resp.json() if is_json else resp.content
+        if _is_json_response(resp):
+            return resp.json()
```

Comment:
Using `BadGordoRequest` means "Represents a general 4xx bad request". I think we need to implement another custom exception for this `BadGordoResponse` as an example
Would be better here:
```python
       try:
           return resp.json()
       except JSONDecodeError as e:
           raise BadGordoResponse(f"Malformed JSON response: {e}")
```